### PR TITLE
Lint conda recipe

### PR DESCRIPTION
**Pull request by jrbourbeau**
This PR adds a new conda recipe linting check (the same one that's run over in conda-forge)

EDIT: I included a temporary commit which should fail the linting check to confirm that the recipe linting in CI is working as expected

### CODE REVIEW

**Comment by jrbourbeau** [comment]
Getting 

```
recipe has some lint:
  ``requirements: run: numba==0.55.1`` must contain a space between the name and the pin, i.e. ``numba ==0.55.1``
```

in the recipe linting job, so things are working as expected

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: Linting
```
Comment (ncclementi) [inline .github/workflows/lint.yaml:1]: Since we are modifying the name, can we make all the extensions match? The other workflow has `yml`

Comment (jrbourbeau) [reply .github/workflows/lint.yaml]: Np, done

**Comment by ncclementi** [comment]
This looks great, going in!

**Pull request merged**